[Bug]: ## Failed to run doctor checks, no dependencies found in package.json

### Describe the bug

## Failed to run doctor checks, no dependencies found in package.json

Encountering an error attempting to perform a storybook migration. Could be a problem with the `keycloakify` project, but the error seems to be coming from `storybook` and at the very least it could create a more descriptive error message to aid in finding the root cause.

### Error message
 
```
Failed to run doctor checks for project ./.storybook:
│  Error: No Storybook dependencies found in the package.json
```

### Reproduction link

https://stackblitz.com/~/github.com/Download/keycloakify-starter

### Reproduction steps

### Stackbliz reproduction

I forked the `keycloakify-starter` project and created a Stackblitz based on that. When you open the link it performs `yarn install` and then starts the dev server. Stop the server with Ctrl+C, then run the migration command: 

```console
npx storybook@latest upgrade
```

Problem is that I can't seem to complete the migration here. It is waiting for me to select the automigrations it should run but it seems to not detect the enter key. When I press enter, a new line is added below the text but nothing else seems to happen... maybe we can use a non-interactive version of the command?

```console
Select automigrations to run
│  ◼ renderer-to-framework (We're moving from renderer-based to framework-based
│    configuration and update your imports and dependencies accordingly.
│    More info: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#mo...)
```

Anyway I added steps below to do this from the terminal and it will probably be quicker that way anyway, so here they go.


### Terminal reproduction

`mkdir` and `cd` into a new empty folder. I called mine `report-storybook-issue`. Then `git clone` the `keycloakify-starter` project, which is using Storybook 8. `npm install` the dependencies, then run the migration. As a part of the migration process it will attempt to run `storybook doctor`, which crashes with this error.

```console
stijn@stijns-mbp ws % mkdir report-storybook-issue
stijn@stijns-mbp ws % cd report-storybook-issue   
stijn@stijns-mbp report-storybook-issue % git clone https://github.com/keycloakify/keycloakify-starter
Cloning into 'keycloakify-starter'...
remote: Enumerating objects: 5156, done.
remote: Counting objects: 100% (636/636), done.
remote: Compressing objects: 100% (116/116), done.
remote: Total 5156 (delta 601), reused 520 (delta 520), pack-reused 4520 (from 4)
Receiving objects: 100% (5156/5156), 12.02 MiB | 40.10 MiB/s, done.
Resolving deltas: 100% (3054/3054), done.
stijn@stijns-mbp report-storybook-issue % cd keycloakify-starter
stijn@stijns-mbp keycloakify-starter % npm install
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated rimraf@2.6.3: Rimraf versions prior to v4 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported

added 873 packages, and audited 874 packages in 21s

206 packages are looking for funding
  run `npm fund` for details

27 vulnerabilities (4 low, 17 moderate, 6 high)

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
stijn@stijns-mbp keycloakify-starter % npx storybook@latest upgrade


┌  Storybook Upgrade - v9.0.13
│
◇  1 project detected
│
●  Upgrading from 8.1.10 to 9.0.13
│
◆  Updated package versions in package.json files
│
◆  1 automigration(s) detected
│
◇  Select automigrations to run
│  renderer-to-framework
│
◆  Completed automigrations for /.storybook
│
◆  Dependencies installed
│
◇  Checking the health of your project(s)..
│
■  Failed to run doctor checks for project ./.storybook:
│  Error: No Storybook dependencies found in the package.json
│
■  /.storybook: 1 problem found
│
│  Configuration Error
│
│  Failed to run doctor checks: No Storybook dependencies found in the package.json
│
◇  Storybook doctor is complete!
│
│  You can always recheck the health of your project(s) by running:
│  npx storybook doctor
│
◇  The upgrade is complete!
│
│  Your project(s) have been upgraded successfully, but some issues were found
│  which need your attention, please check Storybook doctor logs above.
│
│  If you want to learn more about the automigrations that executed in your
│  project(s), please check the following links:
│
│  • renderer-to-framework
│
│  For a full list of changes, please check our migration guide:
│  https://storybook.js.org/docs/migration-guide
│
│  Attention: Storybook now collects completely anonymous telemetry regarding
│  usage. This information is used to shape Storybook's roadmap and prioritize
│  features.
│
│  You can learn more, including how to opt-out if you'd not like to participate in
│  this anonymous program, by visiting the following URL:
│
│  https://storybook.js.org/telemetry
│
│
│
│  Storybook debug logs can be found at: debug-storybook.log
│
└  Done!
```


### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 14.7.3
    CPU: (8) arm64 Apple M1
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.14.0 - ~/.nvm/versions/node/v22.14.0/bin/node
    npm: 11.2.0 - ~/.nvm/versions/node/v22.14.0/bin/npm
    pnpm: 9.1.1 - ~/.nvm/versions/node/v22.14.0/bin/pnpm
  Browsers:
    Chrome: 137.0.7151.120
    Safari: 17.6
  npmPackages:
    @storybook/react-vite: ^9.0.13 => 9.0.13 
    eslint-plugin-storybook: ^9.0.13 => 9.0.13 
    storybook: ^9.0.13 => 9.0.13
```

### Additional context

### Logs

```
stijn@stijns-mbp keycloakify-starter % cat debug-storybook.log
[12:55:53.636] [INFO] Storybook Upgrade - v9.0.13
[12:55:53.644] [INFO] detect-projects-spinner-start: Detecting projects...
[12:55:53.645] [DEBUG] Finding Storybook projects...
[12:55:53.703] [DEBUG] Found 1 Storybook projects
[12:55:53.703] [INFO] detect-projects-spinner: Detecting projects: 1 projects
[12:55:53.704] [DEBUG] Getting Storybook data...
[12:55:53.704] [DEBUG] Getting Storybook info...
[12:55:53.705] [DEBUG] Loading main config...
[12:55:53.782] [DEBUG] Getting stories paths...
[12:55:53.793] [DEBUG] Getting package manager...
[12:55:53.907] [DEBUG] Getting Storybook version...
[12:55:53.908] [DEBUG] /.storybook - Validating before version... 8.1.10
[12:55:53.908] [DEBUG] /.storybook - Validating upgrade compatibility...
[12:55:53.908] [DEBUG] /.storybook - Fetching NPM version information...
[12:55:53.908] [DEBUG] Getting CLI versions from NPM for storybook...
[12:55:53.909] [DEBUG] Getting CLI versions from NPM for storybook@next...
[12:55:54.391] [DEBUG] /.storybook - Evaluating blockers...
[12:55:54.393] [DEBUG] Getting installed version for @storybook/experimental-addon-test...
[12:55:55.104] [INFO] detect-projects-spinner-stop: 1 project detected
[12:55:55.104] [DEBUG] Found 1 valid projects and 0 error projects
[12:55:55.104] [INFO] Upgrading from 8.1.10 to 9.0.13
[12:55:55.109] [INFO] upgrade-dependencies-task-start: Fetching versions to update package.json files..
[12:55:55.109] [DEBUG] Updating dependencies in /.storybook...
[12:55:55.115] [INFO] upgrade-dependencies-task: /package.json
[12:55:55.116] [DEBUG] {
  "upgradedDependencies": []
}
[12:55:55.116] [DEBUG] {
  "upgradedDevDependencies": [
    "storybook@^9.0.13",
    "@storybook/react@^9.0.13",
    "@storybook/react-vite@^9.0.13",
    "eslint-plugin-storybook@^9.0.13"
  ]
}
[12:55:55.116] [DEBUG] {
  "upgradedPeerDependencies": []
}
[12:55:55.116] [DEBUG] Using cached version for storybook...
[12:55:55.116] [DEBUG] Getting CLI versions from NPM for @storybook/react...
[12:55:55.544] [DEBUG] Getting CLI versions from NPM for @storybook/react-vite...
[12:55:56.001] [DEBUG] Getting CLI versions from NPM for eslint-plugin-storybook...
[12:55:56.248] [INFO] upgrade-dependencies-task-success: Updated package versions in package.json files
[12:55:56.249] [INFO] detect-automigrations-task-start: Detecting automigrations...
[12:55:56.249] [DEBUG] Starting automigration collection across 1 projects and 15 fixes...
[12:55:56.253] [INFO] detect-automigrations-task: Checking automigrations for /.storybook...
[12:55:56.253] [DEBUG] Processing project: /.storybook
[12:55:56.253] [DEBUG] Checking fix eslintPlugin for project /.storybook...
[12:55:56.255] [DEBUG] Checking fix addon-mdx-gfm-remove for project /.storybook...
[12:55:56.255] [DEBUG] Checking fix addon-storysource-code-panel for project /.storybook...
[12:55:56.255] [DEBUG] Checking fix upgrade-storybook-related-dependencies for project /.storybook...
[12:55:56.255] [DEBUG] Checking for incompatible storybook packages...
[12:55:56.256] [DEBUG] Checking fix initial-globals for project /.storybook...
[12:55:56.266] [DEBUG] Checking fix addon-a11y-addon-test for project /.storybook...
[12:55:56.266] [DEBUG] Checking fix consolidated-imports for project /.storybook...
[12:55:56.266] [DEBUG] Checking fix addon-experimental-test for project /.storybook...
[12:55:56.266] [DEBUG] Using cached installed version for @storybook/experimental-addon-test...
[12:55:56.266] [DEBUG] Checking fix rnstorybook-config for project /.storybook...
[12:55:56.266] [DEBUG] Checking fix remove-addon-interactions for project /.storybook...
[12:55:56.266] [DEBUG] Checking fix renderer-to-framework for project /.storybook...
[12:55:56.266] [DEBUG] Checking fix remove-essential-addons for project /.storybook...
[12:55:56.267] [DEBUG] Checking fix addon-a11y-parameters for project /.storybook...
[12:55:56.267] [DEBUG] Checking fix remove-docs-autodocs for project /.storybook...
[12:55:56.269] [DEBUG] Checking fix wrap-require for project /.storybook...
[12:55:56.275] [INFO] detect-automigrations-task: 
Automigrations detected:
[12:55:56.280] [INFO] detect-automigrations-task: ✔ renderer-to-framework
[12:55:56.280] [INFO] detect-automigrations-task-success: 1 automigration(s) detected
[12:55:59.506] [PROMPT] Select automigrations to run {"choice":["renderer-to-framework"]}
[12:55:59.506] [INFO] automigrate-/.storybook-task-start: Running automigrations for /.storybook
[12:55:59.507] [DEBUG] 
Migrating @storybook/react to @storybook/react-vite
[12:55:59.509] [DEBUG] Updating package.json files...
[12:55:59.519] [INFO] automigrate-/.storybook-task: ✔ renderer-to-framework
[12:55:59.520] [INFO] automigrate-/.storybook-task-success: Completed automigrations for /.storybook
[12:55:59.520] [INFO] Installing dependencies...
[12:55:59.520] [INFO] install-dependencies-task-start: Installing dependencies...
[12:55:59.645] [INFO] yarn install v1.22.22
[12:55:59.649] [INFO] install-dependencies-task: yarn install v1.22.22
[12:55:59.680] [INFO] [1/5] Validating package.json...
[12:55:59.685] [INFO] install-dependencies-task: [1/5] Validating package.json...
[12:55:59.685] [INFO] [2/5] Resolving packages...
[12:55:59.689] [INFO] install-dependencies-task: [2/5] Resolving packages...
[12:56:00.763] [INFO] [3/5] Fetching packages...
[12:56:00.770] [INFO] install-dependencies-task: [3/5] Fetching packages...
[12:56:03.578] [INFO] [4/5] Linking dependencies...
[12:56:03.588] [INFO] install-dependencies-task: [4/5] Linking dependencies...
[12:56:06.316] [INFO] [5/5] Building fresh packages...
[12:56:06.322] [INFO] install-dependencies-task: [5/5] Building fresh packages...
[12:56:07.043] [INFO] success Saved lockfile.
[12:56:07.049] [INFO] install-dependencies-task: success Saved lockfile.
[12:56:07.049] [INFO] Done in 7.40s.
[12:56:07.055] [INFO] install-dependencies-task: Done in 7.40s.
[12:56:07.062] [INFO] Dependencies installed
[12:56:07.062] [INFO] install-dependencies-task-success: Dependencies installed
[12:56:07.062] [INFO] Checking the health of your project(s)..
[12:56:07.068] [ERROR] Failed to run doctor checks for project ./.storybook:
Error: No Storybook dependencies found in the package.json
[12:56:07.076] [ERROR] /.storybook: 1 problem found
[12:56:07.081] [INFO] Failed to run doctor checks: No Storybook dependencies found in the package.json
[12:56:07.081] [INFO] Configuration Error
[12:56:07.086] [INFO] Failed to run doctor checks: No Storybook dependencies found in the package.json
[12:56:07.092] [INFO] Storybook doctor is complete!
[12:56:07.097] [INFO] You can always recheck the health of your project(s) by running:
npx storybook doctor
[12:56:07.102] [INFO] The upgrade is complete!
[12:56:07.108] [INFO] Your project(s) have been upgraded successfully, but some issues were found which need your attention, please check Storybook doctor logs above.
[12:56:07.118] [INFO] If you want to learn more about the automigrations that executed in your project(s), please check the following links:

• renderer-to-framework
[12:56:07.124] [INFO] For a full list of changes, please check our migration guide: https://storybook.js.org/docs/migration-guide
[12:56:07.129] [INFO] Attention: Storybook now collects completely anonymous telemetry regarding usage. This information is used to shape Storybook's roadmap and prioritize features.
[12:56:07.133] [INFO] You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
[12:56:07.139] [INFO] https://storybook.js.org/telemetry
[12:56:07.144] [INFO] %                                                                                               
```

--- a/code/lib/cli-storybook/src/doctor/getIncompatibleStorybookPackages.test.ts
+++ b/code/lib/cli-storybook/src/doctor/getIncompatibleStorybookPackages.test.ts
@@ -135,6 +135,19 @@ describe('getIncompatibleStorybookPackages', () => {
     vi.mocked(packageManagerMock.getModulePackageJSON).mockReturnValue({ version: '9.0.0' });
   });
 
+  it('succeeds if only core storybook packages used', async () => {
+    vi.mocked(packageManagerMock.getAllDependencies).mockReturnValueOnce({
+      storybook: '9.0.0',
+    });
+
+    const result = await getIncompatibleStorybookPackages({
+      currentStorybookVersion: '9.0.0',
+      packageManager: packageManagerMock as JsPackageManager,
+    });
+
+    expect(result).toHaveLength(0);
+  });
+
   it('returns an array of incompatible packages', async () => {
     // Mock a non-core storybook package that would be found
     vi.mocked(packageManagerMock.getAllDependencies).mockReturnValueOnce({

--- a/code/lib/cli-storybook/src/doctor/getIncompatibleStorybookPackages.ts
+++ b/code/lib/cli-storybook/src/doctor/getIncompatibleStorybookPackages.ts
@@ -1,6 +1,6 @@
 /* eslint-disable local-rules/no-uncategorized-errors */
 import type { JsPackageManager } from 'storybook/internal/common';
-import { versions as storybookCorePackages, versions } from 'storybook/internal/common';
+import { versions as storybookCorePackages } from 'storybook/internal/common';
 import { logger } from 'storybook/internal/node-logger';
 
 import picocolors from 'picocolors';
@@ -102,13 +102,15 @@ export const getIncompatibleStorybookPackages = async (
   context: Context
 ): Promise<AnalysedPackage[]> => {
   const allDeps = context.packageManager.getAllDependencies();
-  const storybookLikeDeps = Object.keys(allDeps).filter(
-    (dep) => dep.includes('storybook') && !versions[dep as keyof typeof versions]
-  );
+  const storybookLikeDeps = Object.keys(allDeps).filter((dep) => dep.includes('storybook'));
   if (storybookLikeDeps.length === 0 && !context.skipErrors) {
     throw new Error('No Storybook dependencies found in the package.json');
   }
-  return Promise.all(storybookLikeDeps.map((dep) => checkPackageCompatibility(dep, context)));
+  return Promise.all(
+    storybookLikeDeps
+      .filter((dep) => !storybookCorePackages[dep as keyof typeof storybookCorePackages])
+      .map((dep) => checkPackageCompatibility(dep, context))
+  );
 };
 
 export const getIncompatiblePackagesSummary = (

